decidim-core/app/packs/src/decidim/account_form.js

```diff
@@ -1,27 +1,73 @@
+/**
+ * Initializes the edit account form to control the password field elements
+ * which should only be required when they are visible.
+ *
+ * @returns {void}
+ */
+const initializeAccountForm = () => {
+  const editUserForm = document.querySelector("form.edit_user");
+  if (!editUserForm) {
+    return;
+  }
+
+  const passwordChange = editUserForm.querySelector("#passwordChange");
+  if (!passwordChange) {
+    return;
+  }
+
+  const passwodFields = passwordChange.querySelectorAll("input[type='password']");
+  if (passwodFields.length < 1) {
+    return;
+  }
+
+  // Foundation uses jQuery so these have to be bound using jQuery and the
+  // attribute value needs to be set through jQuery.
+  const togglePasswordFieldValidators = (enabled) => {
+    $(passwodFields).attr("required", enabled);
+  }
```

Comment:
This would delete input when password fields are hidden, to make sure that form will not be submitted with invisible errors (see video). Also fixes typo in variable name :)

https://user-images.githubusercontent.com/8806781/182412337-ce5c6766-85a6-44eb-9a19-f3f6401ea217.mp4



```suggestion
    const passwordFields = passwordChange.querySelectorAll("input[type='password']");
  if (passwordFields.length < 1) {
    return;
  }

  // Foundation uses jQuery so these have to be bound using jQuery and the
  // attribute value needs to be set through jQuery.
  const togglePasswordFieldValidators = (enabled) => {
    $(passwordFields).attr("required", enabled);

    if (!enabled) {
      $(passwordFields).val("");
    }
  }
```

Reply:
Thanks a lot for the feedback @verarojman !

I did not apply the suggestion directly because it had an indentation error, so I applied it at 5794b25faa0c40244edd6447d28853bef57dfff0. Also, we are trying to get rid of jQuery eventually, so I used plain JS to set the values for the fields according to your suggestion. The only reason for using the `.attr("required")` from jQuery is compatibility with the Foundation Abide plugin that is used in the current layout.

While testing it with your suggestions, I also applied another fix that now shows the password errors for the user in case there were any errors during the update (such as using an old password that you've already used before).

Reply:
Awesome!